### PR TITLE
Add --wrap-threshold option for length-based wrapping (#2340)

### PR DESCRIPTION
**Pull request by sutheesh**
## Summary

Adds a new `--wrap-threshold` option that triggers wrapping of comma-delimited
expressions (function parameters, arguments, and collections) when a line exceeds
the threshold, independently of `--max-width`.

### Motivation

Users frequently want to force-wrap function signatures and collections without
raising `--max-width` globally. A length-based threshold solves this cleanly
without adding per-construct always-wrap flags.

### Behaviour

| Option value | Effect |
|---|---|
| `--wrap-threshold none` (default) | No change — falls back to `--max-width` |
| `--wrap-threshold 80` | Wraps any expression exceeding 80 chars |
| `--wrap-threshold 0` | Always wraps (every multi-argument expression) |

```swift
// --wrap-arguments before-first --max-width 200 --wrap-threshold 40

// Under threshold — left on one line
let result = max(a, b)

// Exceeds threshold — wrapped
func configure(
    title: String,
    subtitle: String,
    icon: Image
) { ... }

### CODE REVIEW

**Comment by nicklockwood** [comment]
@sutheesh looks pretty good! I noticed a couple of edge cases in the test suite though, for example:

```diff
systemUnderTest = TextLinkAdPresenter(
            settings: ANSDKSettings.sharedInstance(),
            textLinkConfiguration: textLinkConfigurationUseCase.textLinkConfiguration(),
            anTextLinkViewFactory: MockANTextLinkAdViewFactoryImplementation(request: request)
-               textLinkConfiguration: textLinkConfigurationUseCase.textLinkConfiguration(),
+               textLinkConfiguration: textLinkConfigurationUseCase.textLinkConfiguration(
+           ),
-           anTextLinkViewFactory: MockANTextLinkAdViewFactoryImplementation(request: request)
+           anTextLinkViewFactory: MockANTextLinkAdViewFactoryImplementation(
+               request: request
+           )
        )
```

I think we should perhaps avoid wrapping functions with < 2 parameters? They'll be wrapped anyway if they exceed `--max-width`, but since `--wrap-threshold` is just an readability limit rather than a hard limit, I think it's more readable if we *don't* wrap these cases.

WDYT?

**Comment by calda** [comment]
I definitely think we should never wrap in parens with no args like `foo()`, I just made a related fix for that in https://github.com/nicklockwood/SwiftFormat/pull/2467.

I'm less opinionated for bodies with a single argument, but see the appeal of excluding those as well.

File: Sources/OptionDescriptor.swift [hunk 1]
```diff
@@ -905,6 +905,15 @@ struct _Descriptors {
         fromArgument: { $0.lowercased() == "none" ? 0 : Int($0).map { max(0, $0) } },
         toArgument: { $0 > 0 ? String($0) : "none" }
     )
+    let wrapThreshold = OptionDescriptor(
+        argumentName: "wrap-threshold",
```
Comment (calda) [inline Sources/OptionDescriptor.swift]: I think we should call the argument something like `list-wrap-threshold` since it only applies to comma-separated lists (right)?

```suggestion
        argumentName: "list-wrap-threshold",
```

File: Sources/OptionDescriptor.swift [hunk 2]
```diff
@@ -905,6 +905,15 @@ struct _Descriptors {
...
+        displayName: "Wrap Threshold",
+        help: "Minimum line length triggering wrap, independent of max-width. Defaults to \"none\"",
```
Comment (calda) [inline Sources/OptionDescriptor.swift]:
```suggestion
        help: "Line length triggering wrap for comma-separated lists, in addition to --max-width. Defaults to \"none\"",
```

**Comment by nicklockwood** [comment]
> I'm less opinionated for bodies with a single argument, but see the appeal of excluding those as well.

Yeah, I'm also OK with leaving them as-is. The empty parens were the more clear cut case.

File: Sources/OptionDescriptor.swift (hunk 1 above)
Comment (sutheesh) [reply Sources/OptionDescriptor.swift]: Make sense addressed the comments

Comment (nicklockwood) [reply Sources/OptionDescriptor.swift]: I was thinking it will be extended in future to include things like SwiftUI operator chains as well. I guess the argument could always be renamed if/when that happens tho.

Comment (sutheesh) [reply Sources/OptionDescriptor.swift]: yeah for now as @calda mentioned kept it as list-wrap-threshold, we can rename it if we plan to include more items in future? @nicklockwood

File: Sources/OptionDescriptor.swift [hunk 3]
```diff
@@ -905,6 +905,15 @@ struct _Descriptors {
         fromArgument: { $0.lowercased() == "none" ? 0 : Int($0).map { max(0, $0) } },
         toArgument: { $0 > 0 ? String($0) : "none" }
     )
+    let listWrapThreshold = OptionDescriptor(
+        argumentName: "list-wrap-threshold",
+        displayName: "List Wrap Threshold",
+        help: "Line length triggering wrap for comma-separated lists, in addition to --max-width. Defaults to \"none\"",
+        keyPath: \.listWrapThreshold,
+        type: .int,
```
Comment (calda) [inline Sources/OptionDescriptor.swift:913]: Can we have the option value be `Int?` so we can use `nil` for `none` instead of `-1`?

File: Tests/Rules/WrapArgumentsTests.swift [hunk 4]
```diff
@@ -3271,4 +3271,79 @@ final class WrapArgumentsTests: XCTestCase {
...
+        )
+        """
+        // listWrapThreshold 0 — wraps multi-argument calls regardless of length
+        let options = FormatOptions(wrapArguments: .beforeFirst, maxWidth: 200, listWrapThreshold: 0)
+        testFormatting(for: input, output, rule: .wrapArguments, options: options)
+    }
+
+    func testWrapThresholdDoesNotWrapSingleArgumentCall() {
+        let input = """
+        let x = someVeryLongFunctionName(someVeryLongArgumentName: someVeryLongValue)
```
Comment (calda) [inline Tests/Rules/WrapArgumentsTests.swift:3316]: I would prefer we wrap single-argument calls over the threshold, just not wrap zero-argument calls like `foo()`.

Based on @nicklockwood's comment here I think he agrees: https://github.com/nicklockwood/SwiftFormat/pull/2510#issuecomment-4315226338

Comment (calda) [reply Tests/Rules/WrapArgumentsTests.swift:3316]: Could you add a test case for a function with zero arguments?

**Review by calda — APPROVED** [review]
